File: 프로그래머스/src/프로그래머스_02272022/정수_제곱근_판별.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    vector<long long> squared(7000000);
    for(long long i = 1 ; i< 7000000; i ++)
    {
        squared[i] = i*i;
    }
 
    int index = 0;
    
    for(long long i = 1;  i < squared.size() ; i ++ )
    {
        if( squared[i] == n )
        {
            index = i;
            break;
        }
        if(squared[i] > n) break;
    }
    
    if(index==0)return -1;
    else return squared[index+1];
    
}

File: 2026/Level1/정수제곱근판별.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12934?language=python3
import math

def solution(n):
    answer = 0
    res = math.sqrt(n)
    if (int(res)**2 == n ) : 
        answer = (res+1)**2
    else :
        answer = -1
    
    return answer